app/uk/gov/hmrc/agentmappingfrontend/controllers/SignedOutController.scala

```diff
@@ -17,21 +17,17 @@
 package uk.gov.hmrc.agentmappingfrontend.controllers
 
 import javax.inject.Inject
-
 import play.api.mvc.{Action, AnyContent}
 import uk.gov.hmrc.agentmappingfrontend.config.AppConfig
+import uk.gov.hmrc.agentmappingfrontend.repository.MappingArnResult.MappingArnResultId
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 
 import scala.concurrent.Future
 
 class SignedOutController @Inject()(appConfig: AppConfig) extends FrontendController {
 
-  def signOut: Action[AnyContent] = Action.async { implicit request =>
-    Future successful Redirect(appConfig.signOutUrl)
-  }
-
-  def signOutAndRedirect: Action[AnyContent] = Action.async { implicit request =>
-    Future successful Redirect(appConfig.signOutAndRedirectUrl)
+  def signOutAndRedirect(id: MappingArnResultId): Action[AnyContent] = Action.async { implicit request =>
+    Future successful Redirect(s"${appConfig.signOutAndRedirectUrl}?id=$id")
```

Comment:
The `?id=$id` part here is not URLEncoded. Perhaps the `FrontendAppConfig.signOutAndRedirectUrl` could be a function that takes this id as parameter? Might be worth seeing how the subscription frontend encodes this.